extra_data/components.py

```diff
@@ -1394,6 +1373,56 @@ def _make_image_index(self, tids, inner_ids, inner_name='pulse'):
         )
 
 
+@multimod_detectors
+class LPD1M(LPDBase, XtdfDetectorBase):
```

Comment:
Does it work for the inheritance to go in a chain (XtdfDetectorBase -> LPDBase -> (LPD1M, LPDMINI)) rather than LPDBase being a mixin class? If it works, I generally find code easier to think about without multiple inheritance involved.

Reply:
Not really: if `LPDBase` inherits from `XtdfDetectorBase`, then it becomes findeable by `identify_multimod_detectors` as a subclass, and breaks the functions that attempt to identify the detector by name and comes up first instead of LPD1M or LPDMini.  

An alternative would be to define the `LPDBase(XtdfDetectorBase)` below the LPD1M and Mini classes :/

Reply:
With the current implementation `identify_multimod_detectors` searches only among classes decorated with `@multimod_detectors`, so if you don't decorate `LPDBase` it will not be findable.

Reply:
Yep, you're right, I misspoke.  
It's [`_iter_trains`](https://github.com/European-XFEL/EXtra-data/blob/master/extra_data/export.py#L72) that iterates through the subclasses and breaks, where `LPDBase` needn't be in the list altogether.

An example is [`test_streamer.test_merge_detector`](https://github.com/European-XFEL/EXtra-data/blob/master/extra_data/tests/test_streamer.py#L13):



```python
mock_fxe_raw_run = '/tmp/tmpqpt42uiw', mock_fxe_control_data = '/tmp/tmpcykco1sb/RAW-R0450-DA01-S00001.h5', mock_spb_proc_run = '/tmp/tmpykz76qep'

    def test_merge_detector(mock_fxe_raw_run, mock_fxe_control_data, mock_spb_proc_run):
        with RunDirectory(mock_fxe_raw_run, include="*LPD[!MINI]*") as run:
>           for tid, data in _iter_trains(run, merge_detector=True):

test_streamer.py:15: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../export.py:72: in _iter_trains
    det = detector(data)
../components.py:1296: in __init__
    super().__init__(data, detector_name, modules, min_modules=min_modules)
../components.py:501: in __init__
    super().__init__(data, detector_name, modules, min_modules=min_modules)
../components.py:113: in __init__
    source_to_modno = self._identify_sources(data, detector_name, modules)
../components.py:176: in _identify_sources
    source_to_modno = dict(self._source_matches(data, detector_name))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'LPDBase' object has no attribute 'detector_name'") raised in repr()] LPDBase object at 0x2b1366583520>, data = <extra_data.DataCollection for 16 sources and 480 trains>, detector_name = 'FXE_DET_LPD1M-1'

    def _source_matches(self, data, detector_name):
        for source in data.instrument_sources:
            m = self._source_re.match(source)
            if m and m.group('detname') == detector_name:
>               yield source, int(m.group('modno'))
E               IndexError: no such group

../components.py:173: IndexError
```

Reply:
I see, thanks. I think probably the right thing to do there is to use `identify_multimod_detectors` instead of `XtdfDetectorBase.__subclasses__`. It's precisely the sort of scenario that function is there for, but we've never got round to looking at it and figuring out if JUNGFRAU needs something slightly different.

However, that bit of refactoring doesn't need to be in this PR either.